### ISSUE TITLE
Upgrade to Unified 10 and Remark 14

### ISSUE THREAD

**Issue by ChristianMurphy** [remcohaszing/remark-mermaidjs#1]
Currently `remark-mermaidjs` is tested with unified 9 and remark 13, ensure it supports unified 10 and remark 14 as well.
https://github.com/remcohaszing/remark-mermaidjs/blob/9c664676214441ef231e53e565b2e8ce413d8aac/package.json#L42
https://github.com/remcohaszing/remark-mermaidjs/blob/9c664676214441ef231e53e565b2e8ce413d8aac/package.json#L71-L72

https://github.com/unifiedjs/unified/releases/tag/10.0.0
https://github.com/remarkjs/remark/releases/tag/14.0.0

The largest challenge would likely be ESM support which impacts a number of tools.
- [ ]  ESM support
   - [ ] configure or replace Jest
   - [ ] configure TypeScript, or convert to JSDoc checked with TypeScript

**Comment by remcohaszing** [comment]
Hey I see you spotted one of my personal remark projects :smile: 

The main reason my personal remark plugins, including this one, still use unified 9, is because I use them in a project where it has to work with Jest, TypeScript, and ts-node. In my experience this still causes issues, but I would love to be proven wrong.

**Comment by ChristianMurphy** [comment]
> Hey I see you spotted one of my personal remark projects

I did!
Expect more traffic in the near future, GitHub is adding Mermaid support, which will likely drive interest https://github.com/github/roadmap/issues/372


> TypeScript, and ts-node

I've had success on my projects using https://github.com/TypeStrong/ts-node/issues/1007

> Jest

It is in theory possible, https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#im-having-problems-with-esm-and-jest gives some pointers on how.
Your mileage may vary, personally, I found migrating off Jest easier. :sweat_smile:
